core/controllers/suggestion.py

```diff
@@ -39,21 +39,105 @@
     """"Handles operations relating to suggestions."""
 
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'suggestion_type': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'target_type': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'target_id': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'target_version_at_submission':
+            {
+                'schema': {
+                    'type': 'int'
+                }
+            },
+            'change': {
+               'schema': {
+                   'type': 'dict',
+                   'properties':
+                    [
+                       {
+                        'name': 'cmd',
+                        'schema': {
+                            'type': 'basestring'
+                        }
+                       },
+                       {
+                        'name': 'state_name',
+                        'schema': {
+                            'type': 'basestring'
+                        }
+                       },
+                       {
+                        'name': 'content_id',
+                        'schema': {
+                            'type': 'basestring'
+                        }
+                       },
+                       {
+                        'name': 'language_code',
+                         'schema': {
+                             'type': 'basestring'
+                         },
+                         'default': None
+                       },
+                       {
+                        'name': 'content_html',
+                        'schema': {
+                            'type': 'html'
+                        }
+                       },
+                       {
+                        'name': 'translation_html',
+                        'schema': {
+                            'type': 'html'
+                        }
+                       },
+                       {
+                        'name': 'data_format',
+                        'schema': {
+                            'type': 'basestring'
+                        }
+                       }
+                    ]
+               }
+            },
+            'description': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            }
+        }
+    }
+
     @acl_decorators.can_suggest_changes
     def post(self):
         """Handles POST requests."""
-        if (self.payload.get('suggestion_type') ==
+        if (self.normalized_payload.get('suggestion_type') ==
                 feconf.SUGGESTION_TYPE_EDIT_STATE_CONTENT):
             raise self.InvalidInputException(
                 'Content suggestion submissions are no longer supported.')
```

Comment:
This part shouldn't be needed if we add list of allowed values for the `suggestion_type`